From: Intel

healthd periodically wakes up to poll health state and perform periodic chores.
It does both uevent based and time based polling.
This causes unwanted wakes from sleep-states.

diff --git a/healthd/healthd.cpp b/healthd/healthd.cpp
index 9b84c3e..b77dbf4 100644
--- a/healthd/healthd.cpp
+++ b/healthd/healthd.cpp
@@ -88,6 +88,7 @@ static int wakealarm_wake_interval = DEFAULT_PERIODIC_CHORES_INTERVAL_FAST;
 static BatteryMonitor* gBatteryMonitor;
 
 static bool nosvcmgr;
+static bool nopolling;
 
 static void wakealarm_set_interval(int interval) {
     struct itimerspec itval;
@@ -282,17 +283,25 @@ int main(int argc, char **argv) {
 
     klog_set_level(KLOG_LEVEL);
 
-    while ((ch = getopt(argc, argv, "n")) != -1) {
+    while ((ch = getopt(argc, argv, "np")) != -1) {
         switch (ch) {
         case 'n':
             nosvcmgr = true;
             break;
+        case 'p':
+            nopolling = true;
+            break;
         case '?':
         default:
             KLOG_WARNING(LOG_TAG, "Unrecognized healthd option: %c\n", ch);
         }
     }
 
+    if (nopolling == true) {
+        healthd_config.periodic_chores_interval_fast = -1;
+        healthd_config.periodic_chores_interval_slow = -1;
+    }
+
     healthd_board_init(&healthd_config);
     wakealarm_init();
     uevent_init();
-- 
2.5.0

